plugin/js/parsers/JadeScrollsParser.js

```diff
@@ -0,0 +1,443 @@
+            // If API fails, fall back to DOM parsing
+            console.error("JadeScrolls API fetch failed, falling back to DOM parsing:", error);
+            let links = [...dom.querySelectorAll("a[href*='/episode-']")];
+            return util.hyperlinksToChapterList(links);
+        }
+    }
+
+    // Extract novel slug from URL
+    extractNovelSlug(url) {
+        let match = url.match(/\/novel\/([^/]+)/);
+        return match ? match[1] : null;
+    }
+
+    // Extract the novel base URL from a chapter URL
+    getNovelUrl(url) {
+        // Convert /novel/{slug}/episode-{n} to /novel/{slug}
+        let match = url.match(/(https?:\/\/[^/]+\/novel\/[^/]+)/);
+        return match ? match[1] : url;
+    }
+
+    // Extract episode/chapter number from URL for sorting
+    extractEpisodeNumber(url) {
+        // Try episode-N format
+        let match = url.match(/episode-(\d+)/);
+        if (match) {
+            return parseInt(match[1], 10);
+        }
+
+        // Try chapter-N format
+        match = url.match(/chapter-(\d+)/);
+        if (match) {
+            return parseInt(match[1], 10);
+        }
+
+        // Fallback: extract any number from the slug
+        match = url.match(/(\d+)/);
+        return match ? parseInt(match[1], 10) : 0;
+    }
+
+    // Extract the chapter title from the page
+    extractChapterTitle(dom) {
+        let titleElement = dom.querySelector(".chpt-title.fw-medium");
+        return titleElement ? titleElement.textContent.trim() : null;
+    }
+
+    // Find the content element on the chapter page
+    findContent(dom) {
+        // Look for the main content container
+        let content = dom.querySelector(".content-inner.listing-disc-styling");
+
+        if (!content) {
+            // Content not found - this happens when the site uses JS to load content
+            // Try to find any content container that might exist
+            content = dom.querySelector("div.content-inner") ||
+                     dom.querySelector("div[class*='content']");
+        }
+
+        return content;
+    }
+
+    // Extract chapter title (called separately from content)
+    findChapterTitle(dom) {
+        let title = this.extractChapterTitle(dom);
+        if (title) {
+            return title;
+        }
+
+        // Fallback: try to get title from page title or h1
+        let h1 = dom.querySelector("h1");
+        if (h1) {
+            return h1.textContent.trim();
+        }
+
+        // Last resort: extract from URL
+        let episodeNum = this.extractEpisodeNumber(dom.baseURI);
+        return episodeNum > 0 ? `Episode ${episodeNum}` : "Chapter";
+    }
+
+    // Get metadata for the current novel from cache
+    getNovelMetadata(dom) {
+        let novelSlug = this.extractNovelSlug(dom.baseURI);
+        return novelSlug ? JadeScrollsParser.metadataCache[novelSlug] : null;
+    }
+
+    // Ensure novel metadata is loaded (lazy loading helper)
+    async ensureNovelMetadata(dom) {
```

Comment:
You should add an override of     async loadEpubMetaInfo(dom) 

and in it call ensureNovelMetadata().  Then you don't need to call ensureNovelMetadata() everywhere.